### PR TITLE
fix: create tool folders with 775 for openshift

### LINKED ISSUE #304: User tool folders have wrong rights
If installing a tool like npm as user, the npm folder has 755 but needs 775 as rights, so the root group can add tools when running with openshift.

- Ref whitesource/renovate-on-prem#193

### PR DESCRIPTION
**Pull request by Chumper**
Closes #304
Closes https://github.com/whitesource/renovate-on-prem/issues/193 after updating.

### CODE REVIEW

**Comment by Chumper** [comment]
@viceice Let me know if root created tool folders should keep 755. 
In that case I can add a line to make a switch

File: src/usr/local/buildpack/utils/filesystem.sh [hunk 1]
```diff
@@ -25,6 +25,8 @@ function find_versioned_tool_path () {
 
 function create_versioned_tool_path () {
   install_dir=$(get_install_dir)
+  # shellcheck disable=SC2174
+  mkdir -p -m 775 "${install_dir}/${TOOL_NAME}"
   mkdir -p "${install_dir}/${TOOL_NAME}/${TOOL_VERSION}"
```
Comment (viceice) [inline src/usr/local/buildpack/utils/filesystem.sh:30]: This needs to be 755 when not running as root too.

Comment (Chumper) [reply src/usr/local/buildpack/utils/filesystem.sh:30]: We can change the global umasks to `755` and `775` and use the refactored methods.
How important is `read+execute` for `other` for you?

If not at all, then we can use the current umasks and simply replace the function.

Comment (viceice) [reply src/usr/local/buildpack/utils/filesystem.sh:30]: I think 770 and 750 are ok.

Comment (Chumper) [reply src/usr/local/buildpack/utils/filesystem.sh:36]: I pushed with 775 and 755 first. Lets keep the legacy install compatible and us 770 and 750 for the new tools going forward.

File: src/usr/local/buildpack/utils/filesystem.sh [hunk 2]
```diff
@@ -25,6 +25,14 @@ function find_versioned_tool_path () {
...
   install_dir=$(get_install_dir)
+
+  local umask=775
+  if [ "$(is_root)" -eq 0 ]; then
+    umask=755
+  fi
+
+  # shellcheck disable=SC2174
+  mkdir -p -m "${umask}" "${install_dir}/${TOOL_NAME}"
   mkdir -p "${install_dir}/${TOOL_NAME}/${TOOL_VERSION}"
```
Comment (viceice) [inline src/usr/local/buildpack/utils/filesystem.sh:36]:
```suggestion
  mkdir -p -m 775 "${install_dir}/${TOOL_NAME}"
   mkdir -p -m "${umask}" "${install_dir}/${TOOL_NAME}/${TOOL_VERSION}"
```
I think this is the right way to go.

Comment (viceice) [reply src/usr/local/buildpack/utils/filesystem.sh]: Ups, that wasn't intended. My intention was to have the version folder 755 on user install.

Misunderstanding, sorry.

File: src/usr/local/buildpack/utils/filesystem.sh [hunk 3]
```diff
@@ -25,7 +25,16 @@ function find_versioned_tool_path () {
...
   install_dir=$(get_install_dir)
-  mkdir -p "${install_dir}/${TOOL_NAME}/${TOOL_VERSION}"
+
+  local umask=775
+  if [ "$(is_root)" -eq 0 ]; then
+    umask=755
+  fi
+
+  # shellcheck disable=SC2174
+  mkdir -p -m 775 "${install_dir}/${TOOL_NAME}"
```
Comment (viceice) [inline src/usr/local/buildpack/utils/filesystem.sh:35]:
```suggestion
  mkdir -p -m "${umask}" "${install_dir}/${TOOL_NAME}"
```
So it should like this? This function is for V1 tools only?

File: test/bash/filesystem.bats [hunk 4]
```diff
@@ -52,6 +54,8 @@ teardown() {
 
     assert_output "${TEST_ROOT_DIR}/root/foo/1.2.3"
     assert [ -d "${TEST_ROOT_DIR}/root/foo/1.2.3" ]
+    assert [ $(stat --format '%a' "${TEST_ROOT_DIR}/root/foo") -eq 775 ]
+    assert [ $(stat --format '%a' "${TEST_ROOT_DIR}/root/foo/1.2.3") -eq 755 ]
```
Comment (viceice) [inline test/bash/filesystem.bats:58]:
```suggestion
    assert [ $(stat --format '%a' "${TEST_ROOT_DIR}/root/foo") -eq 755 ]
    assert [ $(stat --format '%a' "${TEST_ROOT_DIR}/root/foo/1.2.3") -eq 755 ]
```

File: src/usr/local/buildpack/utils/filesystem.sh (hunk 3 above)
Comment (Chumper) [reply src/usr/local/buildpack/utils/filesystem.sh:35]: It is for v1  tools only, but for both - root and user - paths.
Yes, setting the umask in the first `mkdir` only would also be my suggestion. The version should be kept at 755

**Review by viceice — APPROVED** [review]

**Pull request merged**